Proy/game.py

```diff
@@ -1,81 +1,118 @@
+    diags = [[], []]
+    for i in range(3):
+        diags[0].append(game[i][i])
+        diags[1].append(game[i][2 - i])
+    # diags = [[game[i][i] for i in range(3)], [game[i][2 - i] for i in range(3)]]
+    lines = rows + cols + diags
+    for line in lines:
+        if set(line) == {symbol}:  # set([1,1,1]) == {1}
+            return True
+    return False
+
+
+def play_user(game):
+    row, col = get_valid_input(game)
+    update_board(game, row, col, "x")
+    if winner(game, "x"):
+        draw(game)
+        print('\t\t Ganaste "x" !!!')
+        return True
+    return False
+
+
+def play_computer(game):
+    while True:
+        row, col = random.randint(0, 2), random.randint(0, 2)
+        if game[row][col] == " ":
+            update_board(game, row, col, "o")
+            print(f"La Computadora juega: {row}, {col}")
+            if winner(game, "o"):
+                draw(game)
+                print('\t\t Perdiste !!, ganó "o"')
+                return True
+            return False
+
+
 def start():
-    game = [[" "," "," "], [" "," "," "], [" "," "," "]]
-    #game = [[" " for _ in range(3)] for _ in range(3)]
-    sw = 1
-    sw_turn = random.randint(0,1)
+    print("\t Nuevo Juego")
+    game = [[" ", " ", " "], [" ", " ", " "], [" ", " ", " "]]
+    # game = [[" " for _ in range(3)] for _ in range(3)]
+    sw_turn = random.randint(0, 1)
     draw(game)
     for i in range(9):
         if sw_turn == 1:
-            while True:
-                row, col = get_input()
-                if 0 <= row < 3 and 0 <= col < 3: #antes solo verficaba si fuese menor a 3
-                    if game[row][col] == " ":
-                        update_board(game, row, col, "x")
-                        sw_turn = 0
-                        break
-                    print(f"La posicion {row} {col} ya esta ocupada, ingrese otra posicion !!")
-                else:
-                    print("Ingrese una posicion valida existente entre 0 - 2")
-            if winer(game, "x"):
-                for j in range(5):
-                    print('\t\t Ganaste "x" !!!')
-                    draw(game)
-                    time.sleep(1)
-                    os.system("cls")
-                sw = 0
-                print('\t\t Ganaste "x" !!!')
+            sw_turn = 0
+            if play_user(game):
+                return
         else:
-            while True:
-                row, col = random.randint(0,2), random.randint(0,2)
-                if game[row][col] == " ":
-                    update_board(game, row, col, "o")
-                    print("La Computadora juega: ", row, col)
-                    sw_turn = 1
-                    break
-            if winer(game, "o"):
-                print('\t\t Perdiste !!, ganó "o"')
-                sw = 0
+            sw_turn = 1
+            if play_computer(game):
+                return
         draw(game)
-        if sw == 0:
+    print("EMPATE !!")
+
+
+if __name__ == "__main__":
```

Comment:
Two more ideas, if we wanted to extend this program further. Both are not necessary, because Tick-TackToe is such an easy game and you already solved it, as it is playable. Increasing the code coverage would only be of benefit (or even necessary), if we want to extend this with new functionality (e.g. 2d -> 3d, or visualize it better, or make it 4x4 or whatever). If we had that planned, I would insist on improving the code by these suggestions.



1. introduce a game-class that holds the information of the board and coordinates the user/computer (currently, the board is called "game" and it just floats around in the file). It would be natural to make it part of the state of the coordinating class
2. de-duplicate the logic from `play_user` and `play_computer`.  If you think about it, the functions have these steps: 
a) get input
b) update board
c) check if the game is over
They only differ in how they get the input and which symbol they use for b) and c). So strictly, we could factor out the functions `get_user_input` and `get_random_input`